src/data/num/basic.lean

```diff
@@ -216,43 +299,67 @@ end num
 namespace znum
   open pos_num
 
+  /--
+  The negation of a `znum`.
+  -/
   def zneg : znum → znum
   | 0       := 0
   | (pos a) := neg a
   | (neg a) := pos a
 
   instance : has_neg znum := ⟨zneg⟩
 
+  /--
+  The absolute value of a `znum` as a `num`.
+  -/
   def abs : znum → num
   | 0       := 0
   | (pos a) := num.pos a
   | (neg a) := num.pos a
 
+  /--
+  The successor of a `znum`.
+  -/
   def succ : znum → znum
   | 0       := 1
   | (pos a) := pos (pos_num.succ a)
   | (neg a) := (pos_num.pred' a).to_znum_neg
 
+  /--
+  The predecessor of a `znum`.
+  -/
   def pred : znum → znum
   | 0       := neg 1
   | (pos a) := (pos_num.pred' a).to_znum
   | (neg a) := neg (pos_num.succ a)
 
+  /--
+  `bit0 n` appends a `0` to the end of `n`, where `bit0 n = n0`.
+  -/
   protected def bit0 : znum → znum
   | 0       := 0
   | (pos n) := pos (pos_num.bit0 n)
   | (neg n) := neg (pos_num.bit0 n)
 
+  /--
+  `bit1 x` appends a `1` to the end of `x`, mapping `x` to `2 * x + 1`.
+  -/
   protected def bit1 : znum → znum
   | 0       := 1
   | (pos n) := pos (pos_num.bit1 n)
   | (neg n) := neg (num.cases_on (pred' n) 1 pos_num.bit1)
 
+  /--
+  `bitm1 x` appends a `1` to the end of `x`, mapping `x` to `2 * x - 1`.
+  -/
   protected def bitm1 : znum → znum
   | 0       := neg 1
   | (pos n) := pos (num.cases_on (pred' n) 1 pos_num.bit1)
   | (neg n) := neg (pos_num.bit1 n)
 
+  /--
+  Converting an `int` to a `znum`.
```

Comment:
Here are some more starting with "_ing." I'd prefer
```suggestion
  Converts an `int` to a `znum`.
```
Or something written more explicitly if you'd like ("`of_int' z` converts...")